src/common/download.js

```diff
@@ -1,14 +1,9 @@
 export function downloadUrl(url, name, callback) {
   const a = document.createElement('a');
-  a.style.display = 'none';
-  document.body.appendChild(a);
   a.href = url;
-  if (name) a.download = name;
-  a.click();
-  setTimeout(() => {
-    document.body.removeChild(a);
-    if (callback) callback();
-  }, 3000);
+  a.download = name || '';
+  a.dispatchEvent(new MouseEvent('click'));
```

Comment:
Is there any difference between `a.click()` and `a.dispatchEvent(new MouseEvent('click'));`?

Reply:
Yep, click() requires adding to DOM while dispatchEvent works in detached node.

Reply:
Should I add it as a comment?

Reply:
It's OK, we can leave it as is.